misc/json-to-record-converter/README.md

```diff
@@ -0,0 +1,530 @@
+                        <ul>
+                            <li>isString</li>
+                        </ul>
+                    </ul>
+                </td>
+                <td>String</td>
+                <td>string</td>
+            </tr>
+            <tr>
+                <td>
+                    <ul>
+                        <ul>
+                            <li>isNumber (contains &lsquo;.&rsquo;)</li>
+                        </ul>
+                    </ul>
+                </td>
+                <td>Number</td>
+                <td>decimal</td>
+            </tr>
+            <tr>
+                <td>
+                    <ul>
+                        <ul>
+                            <li>isNumber</li>
+                        </ul>
+                    </ul>
+                </td>
+                <td>Integer</td>
+                <td>int</td>
+            </tr>
+            <tr>
+                <td>
+                    <ul>
+                        <li>JsonObject</li>
+                    </ul>
+                </td>
+                <td>Object</td>
+                <td>record</td>
+            </tr>
+            <tr>
+                <td>
+                    <ul>
+                        <li>JsonArray</li>
+                    </ul>
+                </td>
+                <td>Array</td>
+                <td>array</td>
+            </tr>
+        </tbody>
+    </table>
+    <p>&nbsp;</p>
+</div>
+
+### Mapping of Null and Optional Types
+This section explains how different ways an optional Ballerina field can be generated.
+
+#### Scenario 01: JSON null
+When the value of a JSON field is null, it can be mapped as a field of [anydata](https://ballerina.io/spec/lang/master/#anydata) data type in ballerina.
+
+JSON
+```json
+{
+   "nullProperty": null
+}
+```
+Ballerina
+```ballerina
+type NewRecord record {
+   anydata nullProperty;
+};
+```
+
+#### Scenario 02: Optional fields within the same JSON object
+When there are two or more JSON Objects with the same name, and if there are any differencing fields those fields will also be considered as optional fields.
+
+JSON
+```json
+{
+  "object": {
+    "fieldOne": "StringValue",
+    "fieldTwo": 16
+  },
+  "object": {
+    "fieldOne": "StringValue"
+  }
+}
+```
+Ballerina
+```ballerina
+type NewRecord record {
+   string fieldOne;
+   int fieldTwo?;
+};
+```
+
+In the above example, both JSON object fields are with the same name. And there is an intersecting field which is **_fieldOne_** and the differencing field **_fieldTwo_**. Thus, the intersecting field is made to be required and the differencing field is made to be optional.
+
+### Mapping of Primitive Types
+This section explains how JSON primitives can be mapped to Ballerina records. The JSON primitive types are String, Boolean and Number. Any JSON primitive type will be parsed as a **_JsonPrimitive_** Java object. And each object can be checked for String, Boolean or Number by calling the relevant method on the object.
```

Comment:
```suggestion
This section explains how JSON primitives can be mapped to Ballerina records. The JSON primitive types are String, Boolean, and Number. Any JSON primitive type will be parsed as a `_JsonPrimitive_` Java object. Each object can be checked for String, Boolean, or Number by calling the relevant method on the object.
```
We use Oxford commas as a practice in Ballerina documentation. Also, starting sentences with the word `and` is not considered a good writing practice.